devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -642,13 +643,47 @@ void prepare() throws Exception {
             if (devModeContext.isEnablePreview()) {
                 args.add(DevModeContext.ENABLE_PREVIEW_FLAG);
             }
+
+            propagateUserProperties();
+
             args.add("-jar");
             args.add(tempFile.getAbsolutePath());
             if (argsString != null) {
                 args.addAll(Arrays.asList(CommandLineUtils.translateCommandline(argsString)));
             }
         }
 
+        private void propagateUserProperties() {
+            final String mavenCmdLine = BootstrapMavenOptions.getMavenCmdLine();
+            if (mavenCmdLine == null || mavenCmdLine.isEmpty()) {
+                return;
+            }
+            int i = mavenCmdLine.indexOf("-D");
+            if (i < 0) {
+                return;
+            }
+            final StringBuilder buf = new StringBuilder();
+            buf.append("-D");
+            i += 2;
+            while (i < mavenCmdLine.length()) {
+                final char ch = mavenCmdLine.charAt(i++);
+                if (!Character.isWhitespace(ch)) {
```

Comment:
I find this very fragile. IIRC, it's not the first time we discuss about this. What happens if there is a properly escaped space in the property? That happens a lot with paths on Windows.

Can't we get some Maven classes to parse the command line somehow? I thought we did that somewhere else last time we discussed this.

Reply:
You're right about escaping. It's not Maven-specific parsing though.

Reply:
I agree the escaping is not Maven specific but I suppose we should be in line with what Maven does? AFAICS we are in a Maven specific module.

Reply:
I don't think Maven does that anywhere. I can't think of reason why it would need to.

Reply:
I would have expected Maven to have code to parse parameters of the command line? But maybe I'm missing something obvious :).

Reply:
Is there a reason why we need to propogate individual parameters here and not just pass on the full system properties map ?

Reply:
That would probably mean persisting them to a file and then reading it and setting the properties in the dev mode process. I thought it'd be easier to parse a "little" string to pass what's in it on.